rest/data/admin.go

```diff
@@ -0,0 +1,95 @@
+package data
+
+import (
+	"github.com/evergreen-ci/evergreen/model/admin"
+	"github.com/evergreen-ci/evergreen/model/event"
+)
+
+type DBAdminConnector struct{}
+
+// GetAdminSettings retrieves the admin settings document from the DB
+func (ac *DBAdminConnector) GetAdminSettings() (*admin.AdminSettings, error) {
+	return admin.GetSettings()
+}
+
+// SetAdminSettings sets the admin settings document in the DB and event logs it
+func (ac *DBAdminConnector) SetAdminSettings(settings *admin.AdminSettings) error {
+	err := ac.SetAdminBanner(settings.Banner)
+	if err != nil {
+		return err
+	}
+	return ac.SetServiceFlags(settings.ServiceFlags)
+}
+
+// SetAdminBanner sets the admin banner in the DB and event logs it
+func (ac *DBAdminConnector) SetAdminBanner(text string) error {
+	oldSettings, err := admin.GetSettings()
+	if err != nil {
+		return err
+	}
+
+	err = admin.SetBanner(text)
+	if err != nil {
+		return err
+	}
+
+	err = event.LogBannerChanged(oldSettings.Banner, text)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SetServiceFlags sets the service flags in the DB and event logs it
+func (ac *DBAdminConnector) SetServiceFlags(flags admin.ServiceFlags) error {
+	oldSettings, err := admin.GetSettings()
+	if err != nil {
+		return err
+	}
+
+	err = admin.SetServiceFlags(flags)
+	if err != nil {
+		return err
+	}
+
+	err = event.LogServiceChanged(oldSettings.ServiceFlags, flags)
```

Comment:
it seems like, if we're not including some information about the user that is changing the mode, then we should move this into the SetServiceFlags method, so that we always log the state change.

Reply:
Added user to event logging